idx/cassandra/cassandra.go

```diff
@@ -196,17 +205,20 @@ func (c *CasIdx) Add(data *schema.MetricData) error {
 			return err
 		}
 	}
-	if inMemory {
+
+	if inMemory && existing.Partition != partition {
 		oldest := time.Now().Add(-1 * updateInterval).Add(-1 * time.Duration(rand.Int63n(updateInterval.Nanoseconds()*int64(updateFuzzyness*100))))
 		if existing.LastUpdate < oldest.Unix() {
 			log.Debug("cassandra-idx def hasnt been seem for a while, updating index.")
-			existing.LastUpdate = data.Time
-			c.MemoryIdx.AddDef(&existing)
-			c.writeQueue <- writeReq{recvTime: time.Now(), def: &existing}
+			def := schema.MetricDefinitionFromMetricData(data)
+			def.Partition = partition
+			c.MemoryIdx.AddDef(def)
+			c.writeQueue <- writeReq{recvTime: time.Now(), def: def}
```

Comment:
Yes this will handle changes in the partition of a metric. But primarily this is just to persist the change to the "lastUpdate" time.

Previously, we just sent "existing" to the write queue.  But as a result, if the data in cassandra becomes corrupt for any reason it can never be fixed due to metrictank loading the data from cassandra and only every writing that same data back to cassandra with the updated lastUpdate field. 
sending a fresh metricDef built from the metricData is a much more resilient approach and has a negligible performance penalty.